aiida/cmdline/params/options/commands/code.py

```diff
@@ -97,3 +90,29 @@ def is_not_on_computer(ctx):
     prompt_fn=is_on_computer,
     help='Name of the computer, on which the code is installed.'
 )
+
+PREPEND_TEXT = OverridableOption(
+    '--prepend-text',
+    cls=TemplateInteractiveOption,
+    prompt='Prepend script',
+    type=click.STRING,
+    default='',
+    help='Bash code that should be prepended before the executable call in all submit scripts for this code.',
+    extension='.bash',
+    header='**PREPEND_TEXT**\nIf there is any bash code that should be prepended before the executable call in all '
+    'submit scripts for this code, type that between the equal signs below and save the file.',
+    footer='All lines that start with `#=` will be ignored.'
+)
+
+APPEND_TEXT = OverridableOption(
+    '--append-text',
+    cls=TemplateInteractiveOption,
+    prompt='Append script',
+    type=click.STRING,
+    default='',
+    help='Bash code that should be appended after the executable call in all submit scripts for this code.',
+    extension='.bash',
+    header='**APPEND_TEXT**\nIf there is any bash code that should be appended after the executable call in all '
```

Comment:
Careful, this is not prefixed by #= (same for prepend_text)
```
#==========================================================================#
#= **APPEND_TEXT**
If there is any bash code that should be appended after
#= the executable call in all submit scripts for this computer, type that
#= between the equal signs below and save the file.
#==========================================================================#


#==========================================================================#
#= All lines that start with `#=` will be ignored.
#==========================================================================#
```

It might be good to have a test that checks that the stored append_text, if not provided by the user, is actually an empty string.

Reply:
There already is such a test:
https://github.com/aiidateam/aiida-core/blob/f23078e0367aacf7e4e3319ab4ee8e1c48ed6745/tests/cmdline/commands/test_computer.py#L728-L755
It pops the options and runs interactively. The editor just saves the default. You can see that the test checks that the actual values will be the empty string.

Reply:
So why did the test not fail? Or did it fail?

Reply:
It didn't fail. How did you generate the new header? Did you check the code out locally and run `verdi computer setup`? Because when I do it locally (Ubuntu) the newline is properly prefixed with the comment sequence. Might this be a problem of Mac OS? Anyway, to be safe, I have removed the newline in the `header` keyword while updating according to your suggestions. New header looks like this:
```
#==========================================================================#
#= PREPEND_TEXT: if there is any bash commands that should be prepended to the
#= executable call in all submit scripts for this computer, type that
#= between the equal signs below and save the file.
#==========================================================================#
```

look good?

Reply:
> It didn't fail. How did you generate the new header? Did you check the code out locally and run verdi computer setup? 

Yes. I used this yaml
```yaml
label: "daint-mc23"
hostname: "daint.cscs.ch"
description: Piz Daint supercomputer at CSCS Lugano, Switzerland, multicore partition.
transport: ssh
scheduler: "slurm"
work_dir: "/scratch/snx3000/{username}/aiida_run/"
shebang: "#!/bin/bash"
mpirun_command: "srun -n {tot_num_mpiprocs}"
mpiprocs_per_machine: "36"
prepend_text: |
    ### computer prepend_text start ###
    #SBATCH --partition=normal
    #SBATCH --account=mr0
    #SBATCH --constraint=mc
    #SBATCH --cpus-per-task=1
    export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK
    source $MODULESHOME/init/bash
    module load daint-mc
    ulimit -s unlimited
    ### computer prepend_text end ###
#append_text: ""
```
(which correctly prompted me to enter the append text. I added "asda")
```
$ verdi computer setup --config computer.yml
Info: enter "?" for help
Info: enter "!" to ignore the default and set no value
Success: Computer<9> daint-mc23 created
Info: Note: before the computer can be used, it has to be configured with the command:
Info:   verdi computer configure ssh daint-mc23
$ verdi computer show daint-mc23
--------------  -------------------------------------------------------------------------
Label           daint-mc23
PK              9
UUID            39e1450d-d174-4bf2-87e2-538a055e9c66
Description     Piz Daint supercomputer at CSCS Lugano, Switzerland, multicore partition.
Hostname        daint.cscs.ch
Transport type  ssh
Scheduler type  slurm
Work directory  /scratch/snx3000/{username}/aiida_run/
Shebang         #!/bin/bash
Mpirun command  srun -n {tot_num_mpiprocs}
Prepend text    ### computer prepend_text start ###
                #SBATCH --partition=normal
                #SBATCH --account=mr0
                #SBATCH --constraint=mc
                #SBATCH --cpus-per-task=1
                export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK
                source $MODULESHOME/init/bash
                module load daint-mc
                ulimit -s unlimited
                ### computer prepend_text end ###
Append text     If there is any bash code that should be appended after
                asda
--------------  -------------------------------------------------------------------------
```
> Because when I do it locally (Ubuntu) the newline is properly prefixed with the comment sequence.

Reply:
Anyhow, with your latest version, it works fine.

Reply:
I have the suspicion this might be a bug in `click` on MacOS in combination with `jinja2` when using the wordwrap filter. It was due to the literal newline I put in the `{value}` placeholder. Apparently this works on Ubuntu but not on MacOS. Anyway, I removed the newline such that it works everywhere. We'll merge this